tools/cndump/cndb_record.h

```diff
@@ -0,0 +1,126 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#ifndef CNDUMP_CNDB_RECORD_H
+#define CNDUMP_CNDB_RECORD_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <hse/limits.h>
+
+#include <hse_ikvdb/kvs_cparams.h>
+
+#include <cndb/omf.h>
+#include <cn/kvset.h>
+
+struct cndb_rec_version {
```

Comment:
I think we should pull these in-memory structs for cNDB records into HSE lib at some point. That would save us from passing a bunch of output params to all the cNDB omf read interfaces..

Reply:
I had previously made this same comment.

Reply:
It feels like a layer above OMF to me.

Reply:
This feels like an easy way for types and structs to get out of sync with HSE

Reply:
By layer above OMF, I mean in HSE but one step up from OMF because these structs are not the actual OMF.  It would be misleading to define the structs in `*omf.h`.   

With respect to getting out of sync, I 100% agree.  This entire tool has that problem as it's a second parser of OMF (walking KMD entries, mapping them to values that are in vblocks, etc.).  This is partly why I stopped short of pulling the rest of `cn_kbdump` functionality into this tool.  Not entirely sure how to resolve this issue.

Reply:
My point was that having in-memory structs corresponding to the cNDB OMF structs in HSE lib would make the cNDB OMF read interfaces cleaner.

For instance:

```
void
cndb_omf_kvset_add_read(struct cndb_kvset_add_omf *omf, uint64_t *txid, uint64_t *cnid, uint64_t *kvsetid,
                        uint64_t *nodeid, uint64_t *hblkid, unsigned int *kblkc, uint64_t **kblkv,
                        unsigned int *vblkc, uint64_t **vblkv, struct kvset_meta *km)
```
versus
```
void
cndb_omf_kvset_add_read(struct cndb_kvset_add_omf *omf, struct cndb_rec_kvset_add *record, struct kvset_meta *km);
```

Reply:
I agree.  But I don't want to rewrite CNDB right now :)